ui/StatusQ/src/aggregator.cpp

```diff
@@ -0,0 +1,66 @@
+#include "StatusQ/aggregator.h"
+
+Aggregator::Aggregator(QObject *parent)
+    : QObject(parent),
+      m_model(nullptr){
+    connect(this, &Aggregator::modelChanged, this, &Aggregator::calculateAggregation);
+    connect(this, &Aggregator::roleNameChanged, this, &Aggregator::calculateAggregation);
+}
+
+QAbstractItemModel* Aggregator::model() const {
+    return m_model;
+}
+
+void Aggregator::setModel(QAbstractItemModel* model) {
+    if(m_model) {
+        disconnectFromModel();
+    }
+    if (m_model != model) {
+        m_model = model;
+        connectToModel();
+        emit modelChanged();
+    }
+}
+
+QString Aggregator::roleName() const {
+    return m_roleName;
+}
+
+void Aggregator::setRoleName(const QString &roleName) {
+    if (m_roleName != roleName) {
+        m_roleName = roleName;
+        emit roleNameChanged();
+    }
+}
+
+QVariant Aggregator::value() const {
+    return m_value;
+}
+
+bool Aggregator::roleExists() {
+    QHash<int, QByteArray> roles = m_model->roleNames();
+
+    for (const auto& role : roles) {
+        // Check if given role name exists
+        if (QString::fromUtf8(role) == m_roleName) {
+            return true;
+        }
```

Comment:
What about just
```cpp
return m_model->roleNames().contains(m_roleName);
```
?

Reply:
`contains` expects a `key` and our `m_roleName` is a `value`. If there's another easier way, please, share it. That would be more direct, for sure!

Reply:
It could be `return !m_model->roleNames().keys(m_roleName).empty();`